shell/platform/embedder/embedder.cc

```diff
@@ -659,6 +738,16 @@ FlutterEngineResult FlutterEngineInitialize(size_t version,
 
   flutter::Settings settings = flutter::SettingsFromCommandLine(command_line);
 
+  if (SAFE_ACCESS(args, aot_data, nullptr)) {
+    if (SAFE_ACCESS(args, vm_snapshot_data, nullptr) ||
+        SAFE_ACCESS(args, vm_snapshot_instructions, nullptr) ||
+        SAFE_ACCESS(args, isolate_snapshot_data, nullptr) ||
+        SAFE_ACCESS(args, isolate_snapshot_instructions, nullptr)) {
+      return LOG_EMBEDDER_ERROR(kInvalidArguments,
```

Comment:
Please elaborate this error message. Something like "Multiple AOT sources specified. Either specify FlutterProjectArgs::aot_data, or, all of (...)".